src/core/component.js

```diff
@@ -245,6 +251,12 @@ module.exports.registerComponent = function (name, definition) {
   var NewComponent;
   var proto = {};
 
+  if (name.indexOf('__') !== -1) {
+    throw new Error('The component name `' + name + '` is not allowed. ' +
+                    'The sequence __ (double underscore) is reserved and cannot be used ' +
```

Comment:
Would update to `The sequence `__` (double underscore) is reserved for multiple components of the same type.` since  the "cannot be used" is sort of restating the first sentence